source/geometry_model/chunk.cc

```diff
@@ -139,18 +239,101 @@ namespace aspect
-      // during regular operation, but we set manifold_ids for all
-      // cells, faces and edges immediately before refinement and
-      // clear it again afterwards
+      // Attach the real manifold to slot 15.
       coarse_grid.set_manifold (15, manifold);
+      for (typename Triangulation<dim>::active_cell_iterator cell =
+             coarse_grid.begin_active(); cell != coarse_grid.end(); ++cell)
+        cell->set_all_manifold_ids (15);
+
+#if !DEAL_II_VERSION_GTE(9,0,0)
+      // On the boundary faces, set boundary objects.
+      // The east and west boundaries are straight,
+      // the inner and outer boundary are part of
+      // a spherical shell. In 3D, the north and south boundaries
+      // are part of a cone with the tip at the origin.
+      static const StraightBoundary<dim> boundary_straight;
+
+      // Attach boundary objects to the straight east and west boundaries
+      coarse_grid.set_boundary(2, boundary_straight);
+      coarse_grid.set_boundary(3, boundary_straight);
+
+      if (dim == 3)
+        {
+          // Define the center point of the greater radius end of the
+          // north and south boundary cones.
+          // These lie along the z-axis.
+          Point<dim> center;
+          Point<dim> north, south;
+          const double outer_radius = point2[0];
+          north[dim-1] = outer_radius * std::sin(point2[2]);
+          south[dim-1] = outer_radius * std::sin(point1[2]);
+          // Define the radius of the cones
+          const double north_radius = std::sqrt(outer_radius*outer_radius-north[dim-1]*north[dim-1]);
+          const double south_radius = std::sqrt(outer_radius*outer_radius-south[dim-1]*south[dim-1]);
+          static const SpecialConeBoundary<dim> boundary_cone_north(0.0,north_radius,center,north);
+          static const SpecialConeBoundary<dim> boundary_cone_south(0.0,south_radius,center,south);
+          //static const ConeBoundary<dim> boundary_cone_north(0.0,north_radius,center,north);
+          //static const ConeBoundary<dim> boundary_cone_south(0.0,south_radius,center,south);
+
+
+          // Attach boundary objects to the conical north and south boundaries
+          // If one of the boundaries lies at the equator,
+          // just use the straight boundary.
+          if (point2[2] != 0.0)
+            coarse_grid.set_boundary (5, boundary_cone_north);
+          else
+            coarse_grid.set_boundary (5, boundary_straight);
+
+          if (point1[2] != 0.0)
+            coarse_grid.set_boundary (4, boundary_cone_south);
+          else
+            coarse_grid.set_boundary (4, boundary_straight);
+        }
+
+      // Attach shell boundary objects to the curved inner and outer boundaries
+      static const HyperShellBoundary<dim> boundary_shell;
+      coarse_grid.set_boundary (0, boundary_shell);
+      coarse_grid.set_boundary (1, boundary_shell);
+#endif
+    }
+
+#if !DEAL_II_VERSION_GTE(9,0,0)
+    template <int dim>
```

Comment:
unless you have the same `#if` statement in the .h file commenting out the *declaration* of the function.